ast/src/parsed/mod.rs

```diff
@@ -336,10 +336,20 @@ impl<R> Children<Expression<R>> for EnumVariant<Expression<R>> {
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Serialize, Deserialize, JsonSchema)]
 pub struct TraitImplementation<Expr> {
     pub name: SymbolPath,
+    pub source_ref: SourceRef,
     pub type_scheme: TypeScheme,
     pub functions: Vec<NamedExpression<Expr>>,
```

Comment:
What I meant was to use `Arc` here as well, so that we don't need to clone the expression.

Reply:
And then we can make solved_impls "point" at the expression directly and don't need to search for the function or index-access the vector.